Wikipedia/Code/WikidataDescriptionEditingController.swift

```diff
@@ -98,7 +97,23 @@ enum WikidataPublishingError: LocalizedError {
             return
         }
         let requestWithCSRFCompletion: (WikidataAPIResult?, URLResponse?, Error?) -> Void = { result, response, error in
-            completion(result?.error)
+            if let error = error {
+                completion(error)
+            }
+            guard let result = result else {
+                completion(WikidataPublishingError.apiResultNotParsedCorrectly)
+                return
+            }
+
+            completion(result.error)
+
+            // TODO: Check if request was sent with an authorized token instead of checking if user is logged in.
+            let isLoggedIn = WMFAuthenticationManager.sharedInstance.isLoggedIn(at: WMFAuthenticationManager.LoginSite.wikidata)
```

Comment:
Given this CSRFOperation setup, is there a way to know if the request was authorized with a non-anon token in this completion? Looks like the result is the same for anon & non-anon token so can't get it from there. I couldn't think of a good way so resorted to checking if user is logged in

Reply:
Could the completion block pass back the token as well? The token model object might have to be updated with a property that indicates whether it's anon or not